### PR TITLE
feat: expose underlying dataset uri of the table

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**
We frequently need to recover this uri so exposing it to make it easy

### CODE REVIEW

File: rust/lancedb/src/remote/table.rs [hunk 1]
```diff
@@ -325,6 +325,9 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
             message: "table_definition is not supported on LanceDB cloud.".into(),
         })
     }
+    fn dataset_uri(&self) -> &str {
+        "NOT_SUPPORTED"
```
Comment (BubbleCal) [inline rust/lancedb/src/remote/table.rs:329]: throwing an exception could be better?

**Review by BubbleCal — APPROVED** [review]

File: rust/lancedb/src/remote/table.rs (hunk 1 above)
Comment (westonpace) [reply rust/lancedb/src/remote/table.rs:329]: Why can't we support it?  We can just latch the `uri` when the table is constructed.

**Review by westonpace — APPROVED** [review]
This is fine but I think we can support it for remote tables too.

Also, we will eventually want to expose this to python & node I assume

**Comment by chebbyChefNEQ** [comment]
Yeah we def want to start supporting all of these. Just unblocking for the time being will create a ticket

**Pull request merged**